packages/app/src/cli/models/extensions/specifications/transform/app_config_webhook.ts

```diff
@@ -24,12 +24,96 @@ export function transformToWebhookConfig(content: object) {
   const serverWebhooks = getPathValue(content, 'subscriptions') as NormalizedWebhookSubscription[]
   if (!serverWebhooks) return webhooks
 
-  const webhooksSubscriptions: WebhooksConfig['subscriptions'] = []
+  const subscriptions = serverWebhooks.map((subscription) => {
+    const {topic, ...otherFields} = subscription
+    return {...otherFields, topics: [subscription.topic]} as WebhookSubscription
+  })
+  const webhooksSubscriptions: WebhooksConfig['subscriptions'] = mergeAllWebhooks(subscriptions)
 
-  for (const {topic, ...otherFields} of serverWebhooks) {
-    webhooksSubscriptions.push({topics: [topic], ...otherFields})
-  }
-
-  const webhooksSubscriptionsObject = webhooksSubscriptions.length > 0 ? {subscriptions: webhooksSubscriptions} : {}
+  const webhooksSubscriptionsObject = webhooksSubscriptions ? {subscriptions: webhooksSubscriptions} : {}
   return deepMergeObjects(webhooks, {webhooks: webhooksSubscriptionsObject})
 }
+
+export function mergeAllWebhooks(subscriptions: WebhookSubscription[]): WebhookSubscription[] | undefined {
```

Comment:
It might be a good idea to comment where and why this transformer is used... just looking at the above entry point, a lot of these code paths seem impossible. When they are used on the compliance side and initial schema validation, they could be